### PR TITLE
Add integration test for Infrastructure reconciliation + deletion

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
This PR adds a new integration test for the infrastructure controller. Concretely, it creates a namespace + cloud provider secret in a Kubernetes cluster, starts the infrastructure reconciliation, checks whether all resources were correctly created, starts the infrastructure deletion, checks whether all resources are gone again.
The test definition is planned to be executed only during the release pipeline of this components, plus optionally for pull requests where reviewers submitted a `/test` comment (cc @schrodit).

**Special notes for your reviewer**:
It also includes a fix for the VPC gateway endpoints, cc @tareqhs (how did this ever work before?)

**Release note**:

```improvement user
The VPC gateway endpoints provided in the infrastructure configuration are now correctly provisioned.
```
```noteworthy developer
A new integration test for infrastructure reconciliation and deletion has been added.
```

### CODE REVIEW

**Comment by tim-ebert** [comment]
Rebased to do a test run with the tm-bot

File: test/integration/infrastructure/infrastructure_test.go [hunk 1]
```diff
@@ -0,0 +1,1019 @@
...
+
+var _ = Describe("Infrastructure tests", func() {
+	var internalChartsPath string
+
+	BeforeSuite(func() {
+		flag.Parse()
+		validateFlags()
+
+		internalChartsPath = aws.InternalChartsPath
+		aws.InternalChartsPath = filepath.Join("..", "..", "..", aws.InternalChartsPath)
```
[the remark this answers is not in the document]
Comment (schrodit) [reply test/integration/infrastructure/infrastructure_test.go]: The test is executed with `go test -timeout=0 -mod=vendor ./test/integration/infrastructure` therefore go test will automatically set the pwd to `./test/integration/infrastructure`.
It is just different for the gardener suite tests where the wrapper suite resides in a different path.

File: test/integration/infrastructure/infrastructure_test.go [hunk 2]
```diff
@@ -0,0 +1,1019 @@
...
+		restConfig *rest.Config
+		c          client.Client
+
+		scheme        *runtime.Scheme
+		decoder       runtime.Decoder
+		chartRenderer chartrenderer.Interface
+	)
+
+	BeforeEach(func() {
+		awsClient = newAWSClient(*accessKeyID, *secretAccessKey, *region)
```
[the remark this answers is not in the document]
Comment (tim-ebert) [reply test/integration/infrastructure/infrastructure_test.go]: Good point, moved the initialisation code to `BeforeSuite`

**Comment by rfranzke** [comment]
@tim-ebert after the recent changes the integration test is failing, see full logs [here](https://argo.ingress.tm.core.shoot.live.k8s-hana.ondemand.com/workflows/default/bot-test-single-blq7x-wf?tab=workflow&nodeId=bot-test-single-blq7x-wf-408060251&sidePanel=logs%3Abot-test-single-blq7x-wf-408060251%3Amain):

```
STEP: test infrastructure deletion

• Failure [268.809 seconds]

Infrastructure tests

/src/test/integration/infrastructure/infrastructure_test.go:111

  #Reconcile, #Delete

  /src/test/integration/infrastructure/infrastructure_test.go:163

    should correctly create and delete the expected AWS resources [It]

    /src/test/integration/infrastructure/infrastructure_test.go:186

    refusing to compare actual which is not a string

    /src/test/integration/infrastructure/infrastructure_test.go:756
```

Can you please check?

**Comment by tim-ebert** [comment]
Sure, will do.
I ran it locally before pushing yesterday and it completed successfully 🤔

**Comment by tim-ebert** [comment]
Addressed @danielfoehrKn comments and fixed the failing custom matchers, PTAL

File: pkg/aws/matchers/matchers.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+)
+
+// BeSemanticallyEqualTo returns a matcher that tests if actual is semantically
+// equal to the given value from the aws sdk
+func BeSemanticallyEqualTo(expected interface{}) types.GomegaMatcher {
+	if expected == nil {
+		return BeNil()
+	}
+
+	switch expected.(type) {
```
[the remark this answers is not in the document]
Comment (danielfoehrKn) [reply pkg/aws/matchers/matchers.go]: ok got it, thanks!

**Pull request merged**